packages/jest/src/executors/jest/jest.impl.ts

```diff
@@ -153,21 +153,27 @@ export async function batchJest(
   const configPaths = taskGraph.roots.map((root) =>
     path.resolve(context.root, inputs[root].jestConfig)
   );
+  // TODO(caleb): this is depends on the name of the project to also match the 'displayName' in the jest.config.ts
+  //  this can also be an issue if someone has the same project name in a config.
+  //  i.e. two projects will be run for only 1 provided name
+  //  when trying to use readConfigs before runCLI ts throws error about root jest.config.ts.
+  const selectedProjects = taskGraph.roots.map((task) => task.split(':')[0]);
```

Comment:
If you need to squeeze out a little more performance here, you can probably combine these into a single `for...of` loop and just push to separate arrays. 

It's not super important but it's potentially a nice optimization :D